modules/planning/common/speed/st_boundary.cc

```diff
@@ -229,6 +229,46 @@ bool STBoundary::GetBoundarySRange(const double curr_time, double* s_upper,
   return true;
 }
 
+bool STBoundary::GetBoundarySlopes(const double curr_time, double* ds_upper,
+                                   double* ds_lower) const {
+  if (ds_upper == nullptr || ds_lower == nullptr) {
+    return false;
+  }
+  if (curr_time < min_t_ || curr_time > max_t_) {
+    return false;
+  }
+
+  constexpr double kTimeIncrement = 0.05;
+  double t_prev = curr_time - kTimeIncrement;
+  double prev_s_upper = 0.0;
+  double prev_s_lower = 0.0;
+  bool has_prev = GetBoundarySRange(t_prev, &prev_s_upper, &prev_s_lower);
+  double t_next = curr_time + kTimeIncrement;
+  double next_s_upper = 0.0;
+  double next_s_lower = 0.0;
+  bool has_next = GetBoundarySRange(t_next, &next_s_upper, &next_s_lower);
+  double curr_s_upper = 0.0;
+  double curr_s_lower = 0.0;
+  GetBoundarySRange(curr_time, &curr_s_upper, &curr_s_lower);
+  if (!has_prev && !has_next)
+    return false;
```

Comment:
Previous comment allies everywhere in the pull request.

Reply:
Thx. I went through all files of this pr and tried to fix everything that I detected.